### PR TITLE
Enable a clang build

### PR DESCRIPTION
**Pull request by superm1**
Enable a clang build in Debian.

I'm also wanting to do a cross compile using clang, but having various challenges with that.  I'll submit that separately if I get it working.

### CODE REVIEW

File: contrib/ci/README.md [hunk 1]
```diff
@@ -3,32 +3,47 @@ Continuous Integration
...
 
 By using Travis CI, builds are exercised across a variety of environments attempting to maximize code coverage.
-For every commit or pull request 3 builds are performed:
+For every commit or pull request 4 builds are performed:
 
 Fedora
 ------
 
-* A fully packaged RPM build with all plugins enabled
+* A fully packaged RPM build with all plugins enabled\
```
Comment (hughsie) [inline contrib/ci/README.md]: slash?

File: meson.build [hunk 2]
```diff
@@ -117,8 +117,8 @@ add_global_link_arguments(
   language: 'c'
 )
 
-# ONLY for CI, do not enable this for distro builds
-if get_option('enable-werror')
+# ONLY for CI and only when using gcc, do not enable this for distro builds
+if get_option('enable-werror') and meson.get_compiler('c').get_id() == 'gcc'
```
Comment (hughsie) [inline meson.build]: Does clang not support -Werror?

Comment (superm1) [reply meson.build]: It supports `-Werror`, but doesn't support the `-Wnoerror=cpp`

Comment (hughsie) [reply meson.build]: Ahh, okay.

**Comment by hughsie** [comment]
Does `E: Unable to locate package fwupd-dbgsym` mean more to you than me?

**Comment by superm1** [comment]
I think what's going on is that symbols don't get stripped properly when using clang.  So I think that they won't be created in that build.  I've got a test fix to just ignore them.

**Pull request closed without merging**